java/com/google/re2j/RE2.java

```diff
@@ -213,25 +214,43 @@ int numberOfCapturingGroups() {
   // get() returns a machine to use for matching |this|.  It uses |this|'s
   // machine cache if possible, to avoid unnecessary allocation.
   Machine get() {
+    // Treiber stack (if reusing nodes) suffers from ABA problem on pop. Avoid by unlinking the
```

Comment:
Before you get into subtleties, explain what this code does:

// Pop machine from head of toGet free list.
// (Grab a batch from the lock-free stack if empty.)